code/modules/mining/lavaland/necropolis_chests.dm

```diff
@@ -1997,3 +1997,125 @@ GLOBAL_LIST_EMPTY(aide_list)
 			var/mob/living/M = hit_atom
 			if(curse(thrownby, M) == TRUE)
 				to_chat(thrownby, span_notice("You appear before the cane and stab [M], making a new minion out of [M.p_them()]!"))
+
+//Olmec
+
+#define COOLDOWN_HEADCRAFT 10 SECONDS
+/obj/item/claycanister
+	name = "archaic clay canister"
+	desc = "A seemingly bottomless canister of violet clay. Kneading it seems to always results in a shape resembling a head, regardless of intent or concentration."
+	icon = 'yogstation/icons/obj/lavaland/artefacts.dmi'
+	icon_state = "claycan"
+	force = 5 
+	throwforce = 3
+	w_class = WEIGHT_CLASS_NORMAL
+	resistance_flags = LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	var/next_craft = 0
+	var/smashdam = 25 
+	var/animaldam = 50
+	var/objdam = 25
+
+
+/obj/item/claycanister/attack_self(mob/user)
+	if(next_craft > world.time)
+		to_chat(user, span_warning("You can't do that yet!"))
+		return
+	var/obj/item/minihead/B = new()
+	B.origin = src
+	user.put_in_hands(B)
+	to_chat(user, span_notice("You mold a head from the clay!"))
+	next_craft = world.time + COOLDOWN_HEADCRAFT
+
+/obj/item/minihead
+	name = "small clay head"
+	desc = "An earthen homage to its colossal predecessor. The round shape and heft almost scream its desire for flight."
+	icon = 'yogstation/icons/mob/human_parts.dmi'
+	icon_state = "b_golem_head"
+	var/list/headlist = list("abductor_head", "plasmaman_head", "skeleton_head", "cultgolem_head", "fly_head_m", "moth_head_f", "agent_head", "skeleton_head")
+	color = "#774e6d"
+	force = 0
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	var/usedup = FALSE
+	var/obj/item/claycanister/origin
+
+/obj/item/minihead/Initialize(mapload)
+	. = ..()
+	src.icon_state = pick(headlist)
+	
+
+/obj/item/minihead/throw_impact(atom/hit_atom, datum/thrownthing/throwingdatum)
+	. = ..()
+	if(hit_atom && !usedup)
+		origin.follow(hit_atom, 1, src.icon_state)
+		usedup = TRUE
+
+/obj/item/minihead/after_throw(datum/callback/callback)
+	. = ..()
+	if(!usedup)
+		origin.follow(src.loc, 1, src.icon_state)
+		usedup = TRUE
+	qdel(src)
+
+/obj/item/minihead/attack_self(mob/user)
+	if(!usedup)
+		origin.follow(src, 1, src.icon_state)
+		usedup = TRUE
+		qdel(src)
+
+/obj/item/minihead/attack_hand(mob/user)
+	if(isturf(src.loc) && usedup)
+		return
+	. = ..()
+
+/obj/item/claycanister/proc/follow(atom/target, phase = 1, icon_state, var/obj/damocles, var/obj/silhouette, var/turf/landingzone)
```

Comment:
If it always starts at phase 1, just make it different procs rather than one with a switch statement
Easier to keep track of and debug

Reply:
Also, if this is a proc on the jar, not the head, make sure its next to the jar item

Reply:
Also make sure it doesnt runtime if the jar is deleted mid crush